sbi/inference/posteriors/ratio_based_posterior.py

```diff
@@ -107,7 +103,12 @@ def log_prob(
         self._warn_log_prob_snre()
 
         with torch.set_grad_enabled(track_gradients):
-            log_ratio = self.net(torch.cat((theta, x), dim=1)).reshape(-1)
+            # Send to device for evaluation, send to CPU for comparison with prior.
+            log_ratio = (
+                self.net(torch.cat((theta.to(self._device), x.to(self._device)), dim=1))
+                .reshape(-1)
+                .to("cpu")
```

Comment:
Why not return it as `.to(self._device)`? We could do the same for the `.sample()` method, no?

Either way, I think the way `.sample()` and `.log_prob()` deal with the devices should be identical.

Reply:
the call to the `net` happens on the device, but the result has to live on the `cpu` because it will used together with the `prior` which lives on the `CPU`. 
Overall the problem is that only the `net` is on the device, not the prior. The prior comes from the user and we should assume that it just lives on the CPU (open for discussion). I did not find a way to move PyTorch distributions between devices (there is nothing like `to(device)` for them).